lib/helpers.js

```diff
@@ -57,13 +57,24 @@ async function downloadApp (app, appExt) {
 }
 
 async function downloadFile (fileUrl, suffix) {
+
   // We will be downloading the files to a directory, so make sure it's there
   // This step is not required if you have manually created the directory
   let fileInfo = await tempDir.open({prefix: 'appium-app', suffix: suffix});
   _fs.close(fileInfo.fd);
-  await fs.writeFile(fileInfo.path, await request(fileUrl));
-  logger.debug(`${fileUrl} downloaded to ${fileInfo.path}`);
 
+  // we use the non-promise version of request because the 'pipe' function
+  // won't overuse memory. request-promise has to store the whole response
+  // in memory (and also has encoding trickiness)
+  let writeStream = _fs.createWriteStream(fileInfo.path);
+  writeStream.on('error', (e) => logger.errorAndThrow(e));
```

Comment:
You were right @jlipps, it errored, but didn't log correctly the way it was before. Got caught by MJSONWP, but not by BaseDriver.

With the way it works now, it gets logged by BaseDriver

```
info BaseDriver Session created with session id: 4e8f0e21-f0c5-4684-8660-0d49d109c2ab
dbug AndroidDriver Getting Java version
info AndroidDriver Java version is: 1.7.0_67
info BaseDriver Using downloadable app http://appium.s3.amazonaws.com/ApiDemos-debug.apk
ERR! BaseDriver Error: major error
ERR! BaseDriver     at downloadFile$ (lib/helpers.js:75:29)
ERR! BaseDriver     at tryCatch (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/regenerator/runtime.js:67:40)
ERR! BaseDriver     at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/regenerator/runtime.js:294:22)
ERR! BaseDriver     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/regenerator/runtime.js:100:21)
ERR! BaseDriver     at GeneratorFunctionPrototype.invoke (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/regenerator/runtime.js:136:37)
ERR! BaseDriver     at run (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:89:39)
ERR! BaseDriver     at /Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:100:28
ERR! BaseDriver     at doNTCallback0 (node.js:408:9)
ERR! BaseDriver     at process._tickCallback (node.js:337:13)
ERR! BaseDriver  [Error: major error]
dbug AndroidDriver Shutting down Android driver
WARN AndroidDriver Cannot shut down Android driver; it has already shut down
ERR! MJSONWP Encountered internal error running command: Error: Problem downloading app from url http://appium.s3.amazonaws.com/ApiDemos-debug.apk: Error: major error
ERR! MJSONWP     at downloadApp$ (lib/helpers.js:53:11)
ERR! MJSONWP     at tryCatch (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/regenerator/runtime.js:67:40)
ERR! MJSONWP     at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/regenerator/runtime.js:294:22)
ERR! MJSONWP     at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/regenerator/runtime.js:100:21)
ERR! MJSONWP     at GeneratorFunctionPrototype.invoke (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/regenerator/runtime.js:136:37)
ERR! MJSONWP     at run (/Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:89:39)
ERR! MJSONWP     at /Users/jonahss/Workspace/appium-base-driver/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:100:28
ERR! MJSONWP     at doNTCallback0 (node.js:408:9)
ERR! MJSONWP     at process._tickCallback (node.js:337:13)
info HTTP <-- POST /wd/hub/session 500 96.955 ms - 239
```